src/libraries/Microsoft.PowerFx.Core/Binding/INameResolver.cs

```diff
@@ -66,6 +66,15 @@ internal interface INameResolver
         // Looks up the parent control for the current context.
         bool LookupParent(out NameLookupInfo lookupInfo);
 
+        /// <summary>
+        /// In Canvas specifically, this is used to retrieve the full, derived, type
+        /// of a control object where the output schema may be dependent on a control inputs and heirarchy. 
+        /// All Non-Canvas hosts should not implement this method.
+        /// </summary>
+        /// <param name="control">The control to get the full type of.</param>
+        /// <param name="controlType">output: the expanded control type.</param>
+        bool LookupExpandedControlType(IExternalControl control, out DType controlType);
```

Comment:
Why not add a separate interface for this so that it's not forcing other hosts to even think about it?